### ISSUE TITLE
`vmin` and `vmax` don't work as expected when `scheme` is set up

### ISSUE THREAD

**Issue by darribas** [geopandas/geopandas#236]
When a PySAL `scheme` is set up, the values of the series to be plotted are modified in [l.230](https://github.com/geopandas/geopandas/blob/master/geopandas/plotting.py#L230). As a result of that, they are replaced by a sequence [0, k-1]. This renders `vmin` and `vmax` unusable in the way they are thought to be as they won't be in the scale that the new values are set. Just posting it because it took me a while to figure out why it wasn't working, but not sure what the fix would be.

**Comment by jorisvandenbossche** [comment]
On option would be to leave as is (ignore vmin/vmax when `scheme` is passed), but maybe raise a UserWarning if you try to specify vmin/vmax to say they are ignored when also specifying `scheme`.

Otherwise, the `values` could also be truncated based on `vmin`/`vmax` before passing to `__pysal_choro`, although I am not fully sure this would give the desired result.

**Comment by darribas** [comment]
Right, I think at the minimum it'd be informative to pass a UserWarning so it's clear they're not supported.

However, ideally, it'd be great if one would use `vmin` and `vmax` with schemes as they can be useful in some cases (e.g. conditional maps). I think if you included `vmin`/`vmax` in the series to be classified by `__pysal_choro` (and removed them later to pass back the correct set of values), this should take care of problem, but I'm not fully sure. @sjsrey what do you think? Would this be the case?

**Comment by sjsrey** [comment]
@darribas I think there would be other schemes that could be used from within PySAL to accomplish what you want, without having to muck with vmin/vmax directly. If I understand your use case, conditional maps, you would first do a classifier (say quantiles) on the original series. Extract the bins from that classification and apply those bins to the conditional series you want conditional maps for using 
 [User_Defined](https://github.com/pysal/pysal/blob/master/pysal/esda/mapclassify.py#L1450). The conditioned series would thus all have the same classification range, but the distribution of counts across these bins would reflect the conditioning.

But maybe I am misunderstanding your use case?

**Comment by darribas** [comment]
Right, I can see the benefits in terms of simplicity of that approach. I was thinking in a situation like [here](http://darribas.org/gds15/content/labs/lab_04.html#[Extension-2---Advanced]-Conditional-maps) where you're using `seaborn`'s facetting functionality. The easiest I could come up was that approach, where you have a method to generate each of the conditional maps as a choropleth. So if you want to keep the coloring consistent across maps, `vmin` and `vmax` are useful as the classification happens at the subplot level. But there is probably a better way to do it, that involves classifying before facetting.

I like @sjsrey's idea of passing these special cases through a `User_Defined` class. If we had an easy API to plot arbitrary breakpoints (more or less as discussed in Issue #235), I think this would be the way to go because the user could do any kind of classification with all the methods supported by `PySAL` and then pass the output, in whichever context (single maps, subplots, etc.) easily into the plotting framework in `geopandas`. I'm not sure exactly how this would be best implemented, but something that allowed this kind of workflow?

```python
classi = ps.AnyClassification(gdf['y'], **kwds)
gdf.plot('y', scheme='user_defined', bins=classi.bins)
```

I think this would make it extremely flexible with a minimum overhead. What do you think?
